lib/ecto/query/builder.ex

```diff
@@ -218,6 +219,12 @@ defmodule Ecto.Query.Builder do
     error! "`#{Macro.to_string(other)}` is not a valid query expression"
   end
 
+  def runtime_fragment!([{_,_}|_] = kw) when is_list(kw), do: kw
+  def runtime_fragment!(expr) do
+    raise ArgumentError, "to prevent sql injection, only a keyword list may be interpolated " <>
```

Comment:
@lexmag I didn't use `error!/1` as it raises an `Ecto.Query.CompileError` which doesn't seem correct in this case. No?